src/Payments.sol

```diff
@@ -0,0 +1,694 @@
+
+    // railId => Rail
+    mapping(uint256 => Rail) internal rails;
+
+    // token => client => operator => Approval
+    mapping(address => mapping(address => mapping(address => OperatorApproval)))
+        public operatorApprovals;
+
+    // client => operator => railIds
+    mapping(address => mapping(address => uint256[]))
+        public clientOperatorRails;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() public initializer {
+        __Ownable_init(msg.sender);
+        __UUPSUpgradeable_init();
+    }
+
+    function _authorizeUpgrade(
+        address newImplementation
+    ) internal override onlyOwner {}
+
+    modifier validateRailActive(uint256 railId) {
+        require(rails[railId].from != address(0), "rail does not exist");
+        require(rails[railId].isActive, "rail is inactive");
+        _;
+    }
+
+    modifier noRailModificationInProgress(uint256 railId) {
+        require(!rails[railId].isLocked, "Modification already in progress");
+        rails[railId].isLocked = true;
+        _;
+        rails[railId].isLocked = false;
+    }
+
+    modifier onlyRailOperator(uint256 railId) {
+        require(
+            rails[railId].operator == msg.sender,
+            "only the rail operator can perform this action"
+        );
+        _;
+    }
+
+    function approveOperator(
+        address token,
+        address operator,
+        uint256 rateAllowance,
+        uint256 lockupAllowance
+    ) external {
+        require(token != address(0), "token address cannot be zero");
+        require(operator != address(0), "operator address cannot be zero");
+
+        OperatorApproval storage approval = operatorApprovals[token][
+            msg.sender
+        ][operator];
+        approval.rateAllowance = rateAllowance;
+        approval.lockupAllowance = lockupAllowance;
+        approval.isApproved = true;
+    }
+
+    // TODO: Implement Me
+    // https://github.com/FilOzone/payments/pull/1/files#r1974415624
+    function terminateOperator(address operator) external {}
+
+    function deposit(
+        address token,
+        address to,
+        uint256 amount
+    ) external nonReentrant {
+        require(token != address(0), "token address cannot be zero");
+        require(to != address(0), "to address cannot be zero");
+        require(amount > 0, "amount must be greater than 0");
+
+        // Create account if it doesn't exist
+        Account storage account = accounts[token][to];
+
+        // Transfer tokens from sender to contract
+        IERC20(token).safeTransferFrom(msg.sender, address(this), amount);
+
+        // Update account balance
+        account.funds += amount;
+
+        // settle account lockup now that we have more funds
+        settleAccountLockup(account);
+    }
+
+    function withdraw(address token, uint256 amount) external nonReentrant {
```

Comment:
@Stebalien 

I have also implemented a `function withdrawTo(
        address token,
        address to,
        uint256 amount
    )`
    
    function that allows users to withdraw funds to any account of their choosing.